utils/test/src/main/kotlin/Utils.kt

```diff
@@ -99,3 +100,13 @@ fun patchActualResultObject(result: OrtResult, patchStartAndEndTime: Boolean = f
 fun readOrtResult(file: String) = readOrtResult(File(file))
 
 fun readOrtResult(file: File) = file.mapper().readValue<OrtResult>(patchExpectedResult(file))
+
+/**
+ * A helper function to create a custom matcher that compares an [expected] collection to a collection obtained by
+ * [transform] using the provided [matcher].
+ */
+fun <T, U> transformingCollectionMatcher(
+    expected: Collection<U>,
+    matcher: (Collection<U>) -> Matcher<Collection<U>>,
+    transform: (T) -> Collection<U>
+): Matcher<T?> = neverNullMatcher { value -> matcher(expected).test(transform(value)) }
```

Comment:
When talking about "expected" in tests, I like to also talk about "actual". So, would it be appropriate to rename `value` to `actual` here?

Reply:
I'm not sure about this in this case, because the "actual" that is compared is derived from "value", so calling it "actual" could also create confusion in that regard.